plugins/ltac2/tac2intern.ml

```diff
@@ -1845,10 +1851,18 @@ let () =
     let env = List.fold_left fold env ids in
     let loc = tac.loc in
     let (tac, t) = intern_rec env tac in
-    let () = check_elt_unit loc env t in
-    (ist, (ids, tac))
+    let ignore = if List.is_empty ids then match kind env t with
+        | GTypVar _ | GTypRef (Tuple 0, []) -> true
+        | GTypRef (Other kn, []) when KerName.equal kn t_ltac1 -> false
```

Comment:
Dispatching on the type is a very bad design choice. You should instead introduce a new ltac2val quotation.

Reply:
changed